packages/syft/src/syft/core/tensor/smpc/mpc_tensor.py

```diff
@@ -384,42 +384,14 @@ def __apply_op(
         Returns:
             MPCTensor. the operation "op_str" applied on "self" and "y"
         """
+        # relative
+        from .... import parties as mpc_parties
```

Comment:
I don't think the design paradigm of creating a global set of parties that every MPC tensor must exactly adhere to is the paradigm we should be considering. Instead, the list of parties should be attached to the MPCTensor and grow as needed. 

For example - what if I was doing federated learning across 2,000 different data providers? If I wanted to do secure aggregation across clusters of 5 at a time, this design decision would preclude me from doing this (I'd be forced to only and always do SMPC against all 2,000 providers).

Reply:
Modified to Reshare.